src/ripple/conditions/impl/utils.h

```diff
@@ -0,0 +1,384 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPLPE_CONDITIONS_UTILS_H
+#define RIPLPE_CONDITIONS_UTILS_H
+
+#include <ripple/basics/strHex.h>
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include <iomanip>
+#include <sstream>
+#include <utility>
+
+namespace ripple {
+namespace cryptoconditions {
+
+inline
+std::string
+hexstr (std::vector<std::uint8_t> const& data)
+{
+    std::string s;
+    s.reserve (data.size() * 2);
+
+    for (auto d : data)
+    {
+        s.push_back (charHex (d >> 4));
+        s.push_back (charHex (d & 15));
+    }
+
+    return s;
+}
+
+inline
+std::vector<std::uint8_t>
+hexblob (std::string const& s)
+{
+    std::vector<std::uint8_t> tmp;
+    tmp.reserve (1 + (s.size () / 2));
+
+    auto iter = s.cbegin ();
+
+    if (s.size () & 1)
+    {
+        int c = charUnHex (*iter++);
+
+        if (c < 0)
+            Throw<std::runtime_error>("Invalid hex in blob");
```

Comment:
`hexblob` is only used for unit tests. I'll just remove it. The other functions need to be cleaned up.